tests/unit/type/list.tcl

```diff
@@ -1,3 +1,128 @@
+set ::str500 [string repeat x 500000000] ;# 500mb
+
+# Utility function to write big argument into redis client connection
+proc write_big_bulk {size} {
+    r write "\$$size\r\n"
+    while {$size >= 500000000} {
+        r write $::str500
+        incr size -500000000
+    }
+    if {$size > 0} {
+        r write [string repeat x $size]
+    }
+    r write "\r\n"
+    r flush
+    r read
+}
+
+# SORT which attempts to store an element larger than 4GB into a list.
+# Currently unsupported and results in an assertion instead of truncation
+start_server [list overrides [list save ""] ] {
+
+    r config set proto-max-bulk-len 10000000000 ;#10gb
+    r config set client-query-buffer-limit 10000000000 ;#10gb
+
+    test {4gb check push and pop} {
+        set str_length 1000
+        r debug packed_threshold $str_length-1
```

Comment:
@oranagra, I am closing this thread because I split the tests to small / big / compression. and we have more threads regarding other issues with the tests.